### ISSUE TITLE
Document `nested_default_fields` in APIv2

### ISSUE THREAD

**Issue by funkybob** [wagtail/wagtail#11587]
### Pertinent section of the Wagtail docs

https://docs.wagtail.org/en/stable/advanced_topics/api/index.html

(Insert the permalink of the section heading in the Wagtail docs. Choose the most specific section / lowest-level heading to which your suggested change would apply. (In the docs, hover over the heading to see the permalink icon.))

### Details

Whilst the `nested_default_fields` functionality seems well polished, it's not mentioned anywhere in the documentation.

**Comment by arup1221** [comment]
Can I work on it?

**Comment by gasman** [comment]
@arup1221 Yes, anyone is free to work on any issue - no need to ask permission.

**Comment by devv2632** [comment]
Hey there! Is anyone already working on this issue or has it been resolved? If not I would like to give it a try.

**Comment by laymonage** [comment]
@devv2632 no, feel free to go ahead!

**Comment by devv2632** [comment]
Alright! Thank you

**Comment by devv2632** [comment]
Is the documentation for `nested_default_fields` already created and I need to just put it in a proper place or do I have to write the documentation for it too?

**Comment by DeepanshuTevathiya** [comment]
I"ll give it a go!
